n3fit/src/n3fit/model_gen.py

```diff
@@ -634,71 +698,82 @@ def pdfNN_layer_generator(
             )
         )
 
-    # All layers have been made, now we need to connect them,
-    # do this in a function so we can call it for both grids and each replica
-    # Since all layers are already made, they will be reused
-    def compute_unnormalized_pdf(x, neural_network, compute_preprocessing_factor):
+    def neural_network_replicas(x, postfix=""):
```

Comment:
Please add some comments here so that it is a bit more clear what's happening.

(not that the previous comments were that fantastic tbh, but better than nothing)

Reply:
I added some comments, not sure how useful they are. In any case once all the refactorings are done (in particular, I want to also merge the different input grids) the flow should be completely linear, so we won't need to define any functions, we can just create layers and immediately apply them to the output of the previous layers, so it will be much easier to read.